pkg/storage/internal/actions/funder/sql.go

```diff
@@ -161,17 +163,41 @@ func (c *utxoCollector) addToAllocated(utxo *models.UserUTXO) {
 
 func (c *utxoCollector) increaseSize(size uint64) (err error) {
 	c.txSize += size
-	c.result.Fee, err = c.feeCalculator.Calculate(c.txSize)
+	c.fee, err = c.feeCalculator.Calculate(c.txSize)
 	if err != nil {
 		return fmt.Errorf("failed to calculate fee: %w", err)
 	}
 	return nil
 }
 
-func (c *utxoCollector) increaseValue(satoshis uint64) {
+func (c *utxoCollector) increaseValue(sats uint64) error {
+	satoshis, err := to.Int64FromUnsigned(sats)
+	if err != nil {
+		return fmt.Errorf("utxo satoshis value int64: %w", err)
+	}
 	c.satsCovered += satoshis
+	return nil
+}
+
+func (c *utxoCollector) satsToCover() int64 {
+	return c.txSats + c.fee
 }
 
-func (c *utxoCollector) satsToCover() uint64 {
-	return c.txSats + c.result.Fee
+func (c *utxoCollector) prepareResult() (*actions.FundingResult, error) {
+	fee, err := to.UInt64(c.fee)
+	if err != nil {
+		return nil, fmt.Errorf("cannot convert fee to uint64: %w", err)
+	}
+
+	changeAmount, err := to.UInt64(c.satsCovered - c.satsToCover())
+	if err != nil {
+		return nil, fmt.Errorf("cannot convert change amount to uint64: %w", err)
+	}
+
+	return &actions.FundingResult{
+		AllocatedUTXOs: c.allocatedUTXOs,
+		Fee:            fee,
+		ChangeAmount:   changeAmount,
+		ChangeCount:    to.IfThen(changeAmount == 0, 0).ElseThen(1),
+	}, nil
```

Comment:
To be honest, for such a simple condition I would not introduce ternary-operator-like custom method.

```suggestion
	changeCount := 0
	if changeAmount > 0 {
		changeCount = 1
	}

	return &actions.FundingResult{
		AllocatedUTXOs: c.allocatedUTXOs,
		Fee:            fee,
		ChangeAmount:   changeAmount,
		ChangeCount:    changeCount,
	}, nil
```

Reply:
I will take this into account in the final solution,
for now, it's not handling splitting change, so for sure this part will change soon when I finally will tackle the problem of change splitting.